specification/cloudshell/resource-manager/Microsoft.Portal/stable/2018-10-01/CloudShell.json

```diff
@@ -0,0 +1,952 @@
+        "parameters": [
+          {
+            "$ref": "#/parameters/ConsoleNameParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK",
+            "headers": {
+              "x-ms-console-remaining-seconds": {
+                "type": "integer",
+                "description": "Remaining time left in session, in seconds."
+              }
+            }
+          }
+        }
+      }
+    }
+  },
+  "definitions": {
+    "CloudShellConsole": {
+      "description": "Cloud shell console",
+      "type": "object",
+      "required": [
+        "properties"
+      ],
+      "properties": {
+        "properties": {
+          "$ref": "#/definitions/ConsoleProperties"
+        }
+      }
+    },
+    "ConsoleProperties": {
+      "description": "Cloud shell console properties.",
+      "type": "object",
+      "required": [
+        "osType",
+        "provisioningState",
+        "uri"
+      ],
+      "properties": {
+        "osType": {
+          "description": "The operating system type of the cloud shell.",
+          "type": "string",
+          "enum": [
+            "Linux",
+            "Windows"
+          ]
+        },
+        "provisioningState": {
+          "description": "Provisioning state of the console.",
+          "type": "string",
+          "enum": [
+            "NotSpecified",
+            "Accepted",
+            "Pending",
+            "Updating",
+            "Creating",
+            "Repairing",
+            "Failed",
+            "Canceled",
+            "Succeeded"
+          ]
+        },
+        "uri": {
+          "description": "Uri of the console.",
+          "type": "string"
+        }
+      }
+    },
+    "ConsoleCreateProperties": {
+      "description": "Cloud shell properties for creating a console.",
+      "type": "object",
+      "required": [
+        "osType"
+      ],
+      "properties": {
+        "osType": {
+          "description": "The operating system type of the cloud shell.",
+          "type": "string",
+          "enum": [
+            "Linux",
+            "Windows"
+          ]
+        },
+        "provisioningState": {
+          "description": "Provisioning state of the console.",
+          "type": "string",
+          "enum": [
+            "NotSpecified",
+            "Accepted",
+            "Pending",
+            "Updating",
+            "Creating",
+            "Repairing",
+            "Failed",
+            "Canceled",
+            "Succeeded"
+          ]
+        },
+        "uri": {
+          "description": "Uri of the console.",
+          "type": "string"
+        }
+      }
+    },
+    "ConsoleDefinition": {
+      "description": "Console definition",
+      "type": "object",
+      "required": [
+        "properties"
+      ],
+      "properties": {
+        "properties": {
+          "$ref": "#/definitions/ConsoleCreateProperties"
+        }
+      }
+    },
+    "GetUserSettingsResponse": {
```

Comment:
The API responses in this swagger are not conformant with RPC contract for ProxyOnly resources. They are missing the top-level properties: "id","type","name", which are expected for all ARM apis and will actually log provider errors if missing. Is there a compelling reason to exclude them? It should not be a big change as the top level "properties" property is conformant with the resource schema.